### PR TITLE
[Resources and support] Add app for search results

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR implements a new React - the search results page for a new section of the website called Resources and Support, formerly (but still used internally) the Learning Center. It works by -

- There is a build step already for generating "article listings" pages. This PR works off of that to generate a JSON file containing the article data needed to create the search and results.
- The React app fetches the JSON file during boot up and does all the querying client-side

This is an MVP solution and likely won't scale well. We will build a true API during our next iteration. However, for now the JSON file is only about 50 KB, so this seems okay for MVP.

### Todo
- [x] Add prop-types
- [x] Do something about the HTML entities in the results list - `Agent Orange&#x2012;related` 
- [x] Make the search algorithm a little more sophisticated 
- [x] Make the SearchBar component less bad (watch for the "path" parameter during global search)
- [x] Add smoke tests
- [x] Include error state

https://github.com/department-of-veterans-affairs/va.gov-team/issues/14288
https://github.com/department-of-veterans-affairs/va.gov-team/issues/14285
https://github.com/department-of-veterans-affairs/va.gov-team/issues/14478

## Testing done
Tested various queries and pages

### `/resources/search?query=test`

image

### `/resources/search/articles.json`
image

## Acceptance criteria
- [ ] Search results for LC is functional

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by rianfowler — APPROVED** [review]
this application doesn't have a codeowner

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -26,7 +26,7 @@ src/applications/claims-status @department-of-veterans-affairs/vsa-bam-1-fronten
 # Benefits and Memorials 2
 src/applications/disability-benefits/2346 @department-of-veterans-affairs/vsa-bam-2-frontend @department-of-veterans-affairs/vsa-bam-1-frontend @department-of-veterans-affairs/vsa-caregiver-frontend @department-of-veterans-affairs/vsa-ebenefits-frontend
 src/applications/debt-letters @department-of-veterans-affairs/vsa-bam-2-frontend @department-of-veterans-affairs/vsa-bam-1-frontend @department-of-veterans-affairs/vsa-caregiver-frontend @department-of-veterans-affairs/vsa-ebenefits-frontend
-src/applications/financial-status-report @department-of-veterans-affairs/vsa-bam-2-frontend 
+src/applications/financial-status-report @department-of-veterans-affairs/vsa-bam-2-frontend
```
Comment (kelsonic) [inline .github/CODEOWNERS:29]: No space found 🤖

File: src/applications/resources-and-support/components/SearchBar.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+              onChange={event => onInputChange(event.target.value)}
+            />
+          </div>
+          <div className="vads-l-col--12 vads-u-flex--auto vads-u-width--auto">
+            <button
+              type="submit"
+              className="usa-button vads-u-margin--0 vads-u-width--full vads-u-width--auto vads-u-height--full"
+              style={{ borderTopLeftRadius: 0, borderBottomLeftRadius: 0 }}
+            >
+              <i className="fa fa-search"> </i> Search
```
[the remark this answers is not in the document]
Comment (kelsonic) [reply src/applications/resources-and-support/components/SearchBar.jsx:93]: do we want `role="presentation"` here?

File: src/applications/resources-and-support/tests/SearchResultList.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        ],
+      },
+    },
+  },
+];
+
+describe('SearchResultList', () => {
+  it('renders without crashing', () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<SearchResultList results={results} />, div);
```
Comment (kelsonic) [inline src/applications/resources-and-support/tests/SearchResultList.unit.spec.jsx:64]: How come you didn't use `mount` or `shallow` instead?

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉 We can write tests later for this, really nice job!

File: src/applications/resources-and-support/tests/SearchResultList.unit.spec.jsx (hunk 3 above)
Comment (ncksllvn) [reply src/applications/resources-and-support/tests/SearchResultList.unit.spec.jsx:64]: I could have but I just went with what the Create React App docs said for quick and easy smoke tests

**Pull request merged**